test/scripts/cloud-prepare/test.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
+
+source ${SCRIPTS_DIR}/lib/debug_functions
+source ${SCRIPTS_DIR}/lib/utils
+
+function verify_gw_topology() {
+    local expected_gateways="${1}"
+    local gateways_count
+
+    echo "Verifying ${expected_gateways} gateway nodes"
+    gateways_count=$(kubectl get nodes -l=submariner.io/gateway=true \
+        -o yaml | yq '.items | length' -)
+
+    if [[ "$expected_gateways" -ne "$gateways_count" ]]; then
+        echo "Expected ${expected_gateways} gateways nodes but detected ${gateways_count}"
+        return 1
+    fi
+
+    echo "Found expected number of gateways - ${expected_gateways}"
+}
+
+function remove_gw_labels() {
+    local gw_nodes
+
+    echo "Reset gateway nodes (unlabel)"
+    readarray -t gw_nodes < <(kubectl get nodes -l=submariner.io/gateway=true \
+        -o yaml | yq '.items[].metadata.name' -)
+
+    for node in "${gw_nodes[@]}"; do
+        kubectl label --overwrite nodes "$node" submariner.io/gateway-
+    done
+}
+
+echo "Prepare cluster"
+make clusters SETTINGS="$(dirname $0)/no_gw_defined.yml"
+declare_kubeconfig
+
+for scenario in no_gw_defined.yml two_gw_defined.yml; do
+    export SETTINGS="$(dirname $0)/${scenario}"
+
+    test_gateways=$(yq e '.gateways' $SETTINGS)
```

Comment:
This will only work if `gateways` is defined on the global level not on cluster level.
Also its still prone to reading from settings file, which is more fragile.

I propose to just call the test with hard coded values (1 and 2) instead of relying on reading from settings.
You can wrap L46-L51 in a function `test_gateways` or something like that

Reply:
@mkolesnik 
But all the purpose of this test is to ensure that the configuration could be properly read from the config file.
That's why I think it's preferred to use the previous way of reading from file:

`local gw_count="${cluster_gateways[$cluster]:-1}"`

It's the most convenient as uses the actual deployment flow.

Reply:
When testing programatically it's better not to reuse the same underlying mechanism if possible, to avoid using a broken mechanic.

Say if I put now `gateways: 0` in the two gw file, the test will still pass, but the printout won't really help us understand the test is broken.

Since you know how many gateways you expect in each scenario, no reason not to provide it as a constant value.

Reply:
@mkolesnik 

I'm not sure I'm following your suggestion.
You said that I can wrap L46-L51 in a function test_gateways or something like that.

Now, looking at the code.
The loop here (line 40), applies and test two scenarios.
The "make cloud-prepare" (line 47) applies the settings from the scenario files.
I don't touch it, since it's the actual test of the functionality.

So, you would like to hardcode the input values for the "verify_gw_topology" test in line 50.
But since it's running as part of the loop, I can't really change the hardcore values.